### PR TITLE
Migrate pallet-im-online benchmark to v2

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Part of:

- #6202.

### CODE REVIEW

**Review by ggwpez — APPROVED** [review]

File: substrate/frame/im-online/src/benchmarking.rs [hunk 1]
```diff
@@ -19,17 +19,15 @@
 
 #![cfg(feature = "runtime-benchmarks")]
 
-use super::*;
-
-use frame_benchmarking::v1::benchmarks;
+use frame_benchmarking::{v2::*, BenchmarkError};
```
Comment (re-gius) [inline substrate/frame/im-online/src/benchmarking.rs:22]:
```suggestion
use frame_benchmarking::v2::*;
```

`BenchmarkError` should already be in v2

File: substrate/frame/im-online/src/benchmarking.rs [hunk 2]
```diff
@@ -64,34 +62,55 @@ pub fn create_heartbeat<T: Config>(
...
-	validate_unsigned {
-		let k in 1 .. MAX_KEYS;
-		let (input_heartbeat, signature) = create_heartbeat::<T>(k)?;
+#[benchmarks]
+mod benchmarks {
+	use super::*;
+
+	#[benchmark(extra)]
+	fn heartbeat(k: Linear<1, MAX_KEYS>) -> Result<(), BenchmarkError> {
+		let (input_heartbeat, signature) = create_heartbeat::<T>(k.into())?;
```
Comment (re-gius) [inline substrate/frame/im-online/src/benchmarking.rs:71]:
```suggestion
		let (input_heartbeat, signature) = create_heartbeat::<T>(k)?;
```
Are you sure that `into()` is actually needed? It shouldn't be, the same applies below

**Review by shawntabrizi — APPROVED** [review]

File: substrate/frame/im-online/src/benchmarking.rs [hunk 3]
```diff
@@ -64,34 +62,55 @@ pub fn create_heartbeat<T: Config>(
...
+
+		Ok(())
 	}
 
-	validate_unsigned_and_then_heartbeat {
-		let k in 1 .. MAX_KEYS;
-		let (input_heartbeat, signature) = create_heartbeat::<T>(k)?;
+	#[benchmark]
+	fn validate_unsigned_and_then_heartbeat(k: Linear<1, MAX_KEYS>) -> Result<(), BenchmarkError> {
+		let (input_heartbeat, signature) = create_heartbeat::<T>(k.into())?;
```
Comment (re-gius) [inline substrate/frame/im-online/src/benchmarking.rs:95]:
```suggestion
		let (input_heartbeat, signature) = create_heartbeat::<T>(k)?;
```

Comment (re-gius) [reply substrate/frame/im-online/src/benchmarking.rs:95]: Applying also to this last occurrence left

**Pull request merged**